frame/nis/src/lib.rs

```diff
@@ -224,7 +226,8 @@ pub mod pallet {
 			+ Default
 			+ From<u64>
 			+ TypeInfo
-			+ MaxEncodedLen;
+			+ MaxEncodedLen
+			+ CheckedNeg;
```

Comment:
Why is this necessary? This PR does not remove any bounds in this pallet. Additionally, this type seems to have a lot of repeated bounds over `trait Balance` through which it is already bound in the `type Currency` above.

Reply:
`CurrencyBalance` is just the `Balance` type + `From<u64>` and therefore it needs to implement `Balance`. Todo it through its blanket impl now we need, the newly added, `CheckedNeg`.
Alternatively, to make it more clear, I guess we could just do 
```rust
type CurrencyBalance: Balance + From<u64>;
```

Reply:
Got it. Yes. I think your suggestion is good. It would bring it more in line with the doc on it.